shiny/workflowPlot/server.R

```diff
@@ -3,99 +3,198 @@ library(PEcAn.DB)
+    split_string <- strsplit(diff_ids,',')[[1]]
+    # Workflow id is the first element. Trim leading and ending white spaces. Split by space now
+    wID <- as.numeric(strsplit(trimws(split_string[1],which = c("both")),' ')[[1]][2])
+    # Run id is the second element
+    runID <- as.numeric(strsplit(trimws(split_string[2],which = c("both")),' ')[[1]][2])
+    return(data.frame(wID,runID))
+  }
+  # Wrapper over return_DF_from_run_ID
+  # @param list of multiple run ids
+  # run_id_string: ('workflow' workflow_ID, 'run' run_id)
+  # @return Data Frame of workflow and run ids
+  parse_ids_from_input_runID <- function(run_id_list){
+    globalDF <- data.frame()
+    for(w_run_id in run_id_list){
+      globalDF <- rbind(globalDF,return_DF_from_run_ID(w_run_id))
+    }
+    return(globalDF)
+  }  
+  # Fetches variable names from DB
+  # @param workflow_id and run_id
+  # @return List of variable names
+  var_names_all <- function(workflow_id, run_id){
+    # Get variables for a particular workflow and run id
+    var_names <- get_var_names(bety, workflow_id, run_id)
+    # Remove variables which should not be shown to the user
+    removeVarNames <- c('Year','FracJulianDay')
+    var_names <- var_names[!var_names %in% removeVarNames]
+    return(var_names)
+  }
+  # Update variable names  
   observe({
-    ignore <- input$variable_name
-    ranges$x <- NULL
-    ranges$y <- NULL
-  })
-
-  observeEvent(input$plot_dblclick, {
-    brush <- input$plot_brush
-    if (!is.null(brush)) {
-      ranges$x <- as.POSIXct(c(brush$xmin, brush$xmax), origin = "1970-01-01", tz = "UTC")
-      ranges$y <- c(brush$ymin, brush$ymax)
-    } else {
-      ranges$x <- NULL
-      ranges$y <- NULL
+    req(input$all_run_id)
+    # All information about a model is contained in 'all_run_id' string
+    ids_DF <- parse_ids_from_input_runID(input$all_run_id)
+    var_name_list <- c()
+    for(row_num in 1:nrow(ids_DF)){
+      var_name_list <- c(var_name_list,var_names_all(ids_DF$wID[row_num],ids_DF$runID[row_num]))
     }
+    updateSelectizeInput(session, "variable_name", choices=var_name_list)
   })
-
-  output$outputPlot <- renderPlot({
-    workflow_id <- isolate(input$workflow_id)
-    run_id <- isolate(input$run_id)
-    var_name <- input$variable_name
-    if (workflow_id != "" && run_id != "" && var_name != "") {
-      workflow <- collect(workflow(bety, workflow_id))
-      if(nrow(workflow) > 0) {
-        outputfolder <- file.path(workflow$folder, 'out', run_id)
-        files <- list.files(outputfolder, "*.nc$", full.names=TRUE)
-        dates <- NA
-        vals <- NA
-        title <- var_name
-        ylab <- ""
-        for(file in files) {
-          nc <- nc_open(file)
+  # Load data for a single run of the model
+  # @param workflow_id and run_id
+  # @return Dataframe for one run 
+  # For a particular combination of workflow and run id, loads
+  # all variables from all files. 
+  load_data_single_run <- function(workflow_id,run_id){
```

Comment:
Adding in [query.dplyr.R](https://github.com/PecanProject/pecan/blob/develop/db/R/query.dplyr.R). Is that good?